pontoon/base/models/entity.py

```diff
@@ -870,21 +883,20 @@ def for_project_locale(
 
             # Search in source strings
             if not search_translations_only:
-                # Search in string (context) identifiers
-
-                q_key = Q()
-                # TODO: Uncomment the 5 lines below to reactivate the
-                #       context identifiers filter once .ftl bug is fixed (issue #3284):
-                # q_key = (
-                #     Q(**{f"key__{case_lookup}": (search)})
-                #     if search_identifiers
-                #     else Q()
-                # )
-
                 entity_filters = (
-                    Q(**{f"string__{i}regex": rf"{y}{s}{y}"})
-                    | Q(**{f"string_plural__{i}regex": rf"{y}{s}{y}"})
-                    | q_key
+                    Q(
+                        Q(resource__format="ftl")
+                        & (Q(**{f"string__{i}regex": rf"{r}{y}{s}{y}{o}"}))
+                    )
+                    | Q(
+                        ~Q(resource__format="ftl")
+                        & Q(**{f"string__{i}regex": rf"{y}{s}{y}"})
```

Comment:
For non-FTL strings (which include PO strings) we need to search in `string_plural`.